File: cmd/billing.go
package cmd

import (
	"fmt"

	"github.com/CraftIntertech/craft-cli/internal/client"
	"github.com/CraftIntertech/craft-cli/internal/output"
	"github.com/spf13/cobra"
)

var billingShowCmd = &cobra.Command{
	Use:   "show [VM_ID]",
	Short: "Show billing information for a VM",
	Run: func(cmd *cobra.Command, args []string) {
		vmID := ""
		if len(args) > 0 {
			vmID = args[0]
		}
		if vmID == "" {
			vmID = SelectVM("Select VM for billing")
		}

		result := client.Get(fmt.Sprintf("/vms/%s/billing", vmID), nil, true)
		output.PrintItem(extractObj(result))
	},
}

var billingRenewCmd = &cobra.Command{
	Use:   "renew [VM_ID]",
	Short: "Renew a VM billing cycle",
	Run: func(cmd *cobra.Command, args []string) {
		vmID := ""
		if len(args) > 0 {
			vmID = args[0]
		}
		if vmID == "" {
			vmID = SelectVM("Select VM to renew")
		}

		billingType, _ := cmd.Flags().GetString("billing-type")
		if billingType == "" {
			billingType = SelectBillingType("Select billing type")
		}

		body := map[string]interface{}{
			"billingType": billingType,
		}

		result := client.Post(fmt.Sprintf("/vms/%s/renew", vmID), body, true, 0)
		output.PrintSuccess("VM renewed successfully.")
		output.PrintItem(extractObj(result))
	},
}

var billingAutoRenewCmd = &cobra.Command{
	Use:   "auto-renew [VM_ID]",
	Short: "Enable or disable auto-renewal for a VM",
	Run: func(cmd *cobra.Command, args []string) {
		vmID := ""
		if len(args) > 0 {
			vmID = args[0]
		}
		if vmID == "" {
			vmID = SelectVM("Select VM for auto-renew")
		}

		enable, _ := cmd.Flags().GetBool("enable")
		disable, _ := cmd.Flags().GetBool("disable")

		var autoRenew bool
		if enable {
			autoRenew = true
		} else if disable {
			autoRenew = false
		} else {
			autoRenew = Confirm("Enable auto-renewal?")
		}

		body := map[string]interface{}{
			"autoRenew": autoRenew,
		}

		client.Patch(fmt.Sprintf("/vms/%s/auto-renew", vmID), body, true)
		if autoRenew {
			output.PrintSuccess("Auto-renewal enabled.")
		} else {
			output.PrintSuccess("Auto-renewal disabled.")
		}
	},
}

func init() {
	billingRenewCmd.Flags().String("billing-type", "", "Billing type (daily, weekly, monthly, yearly)")

	billingAutoRenewCmd.Flags().Bool("enable", false, "Enable auto-renewal")
	billingAutoRenewCmd.Flags().Bool("disable", false, "Disable auto-renewal")

	BillingCmd.AddCommand(billingShowCmd)
	BillingCmd.AddCommand(billingRenewCmd)
	BillingCmd.AddCommand(billingAutoRenewCmd)
}

File: craft/commands/billing.py
import click

from craft.client import get, patch, post
from craft.output import print_item, print_success


@click.command("show")
@click.argument("vm_id", required=False, default=None)
def billing_show(vm_id):
    """Get VM billing status."""
    if not vm_id:
        from craft.interactive import select_vm
        vm_id = select_vm()
    data = get(f"/vms/{vm_id}/billing")
    print_item(data)


@click.command("renew")
@click.argument("vm_id", required=False, default=None)
@click.option("--billing-type", default=None, type=click.Choice(["daily", "weekly", "monthly", "yearly"]))
def billing_renew(vm_id, billing_type):
    """Renew VM for another period."""
    if not vm_id:
        from craft.interactive import select_vm
        vm_id = select_vm()
    if not billing_type:
        from craft.interactive import select_billing_type
        billing_type = select_billing_type()
    data = post(f"/vms/{vm_id}/renew", {"billingType": billing_type})
    print_success("VM renewed.")
    print_item(data)


@click.command("auto-renew")
@click.argument("vm_id", required=False, default=None)
@click.option("--enable/--disable", default=None, help="Enable or disable auto-renew")
def billing_auto_renew(vm_id, enable):
    """Toggle VM auto-renewal."""
    if not vm_id:
        from craft.interactive import select_vm
        vm_id = select_vm()
    if enable is None:
        from craft.interactive import confirm
        enable = confirm("Enable auto-renew?")
    patch(f"/vms/{vm_id}/auto-renew", {"autoRenew": enable})
    state = "enabled" if enable else "disabled"
    print_success(f"Auto-renew {state}.")
